accepted/0000-prometheus-tls.md

```diff
@@ -0,0 +1,238 @@
+
+## Certificates generation
+
+### Prometheus server
+
+#### Generate a Prometheus server key and request for signing (CSR)
+
+```
+openssl genrsa -out prometheus-server.key 2048
+openssl req -sha256 -text -config server/rhn-prometheus-server-openssl.cnf
+    -new -key prometheus-server.key -out prometheus-server.csr
+```
+
+#### Sign a certificate with CA
+
+```
+openssl ca -extensions req_server_x509_extensions -outdir ./
+    -config rhn-ca-openssl.cnf -in prometheus-server.csr -batch
+    -cert RHN-ORG-TRUSTED-SSL-CERT -keyfile RHN-ORG-PRIVATE-SSL-KEY
+    -days 365 -md sha256 -out prometheus-server.crt
+```
+
+> By setting `x509_extensions.extendedKeyUsage = serverAuth, clientAuth` in
+> openSSL configuration file, the certificate can be used both as the server 
+> certificate and as client certificate (for authenticating scraping requests).
+
+### Minions (Exporter nodes)
+
+The procedure for generating certificates and private keys for minions is the
+same as for the Prometheus server. During onboarding and hostname change the
+generated certificates should be distributed to affected minions.
+
+Recreating CA certificate requires updating all server certificates. This could
+be avoided if we decide to use basic auth with self-signed certificates (not CA
+signed).
+
+## Example configuration
+### Prometheus server
+
+`prometheus.yml`
+```yml
+global:
+  scrape_interval: 15s
+  evaluation_interval: 15s
+
+scrape_configs:
+  - job_name: 'prometheus'
+    scheme: https
+    basic_auth:
+      username: witek
+      password: bedyk
+    tls_config:
+      ca_file: prometheus-server.crt
+    static_configs:
+      - targets: ['minion-1.tf.local:9090']
+
+  - job_name: 'node'
+    scheme: https
+    tls_config:
+      ca_file: /etc/pki/trust/anchors/RHN-ORG-TRUSTED-SSL-CERT  # CA certificate
+      cert_file: prometheus-server.crt                          # client certificate
+      key_file: prometheus-server.key                           # client private key
+    static_configs:
+      - targets: ['minion-2.tf.local:8080']
+```
+
+`web.yml`
+```yml
+tls_server_config:
+  cert_file: prometheus-server.crt    # server certificate
+  key_file: prometheus-server.key     # server private key
+basic_auth_users:
+  witek: $2y$10$vMoFFaIhSWxSBYQSAtaUReypiaovz2Gb.UWhYL1BSv58Vfdxn55jG
+```
+
+### Node exporter
+
+`web.yml`
+
+```yml
+tls_server_config:
+  cert_file: minion.crt                   # server certificate
+  key_file: minion.key                    # server private key
+  client_ca_file: /etc/pki/trust/anchors/RHN-ORG-TRUSTED-SSL-CERT # CA used for signing client/server certificate on Prometheus server
+  client_auth_type: RequireClientCert     # require client certificate for authentication
+```
+
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+One important drawback is the increased complexity of deployment. In
+particular changing client hostnames or renewing CA certificate will require
+updating the certificates on clients.
```

Comment:
Maybe we can leverage the cert-formula for deployment? https://github.com/saltstack-formulas/cert-formula
Given Uyuni is a systems management solution, it should manage :-) (although I agree certificates are usually generated by different departments which may have their own means for distribution)

Reply:
Added to the references list.